src/generic_core/social.spec.ts

```diff
@@ -339,7 +337,7 @@ describe('Social.FreedomNetwork', () => {
 
   it('JSON.parse and stringify messages at the right layer', (done) => {
     var user = network.getUser('mockuser');
-    spyOn(user, 'handleMessage');
+    spyOn(user, 'handleMessage').and.returnValue(Promise.resolve());
```

Comment:
Why does this need to return Promise.resolve()?  The real handleMessage methods now just return void, and it looks like the test cases have been changed to reflect that